aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -498,6 +630,69 @@ void gemv<at::BFloat16>(CUDABLAS_GEMV_ARGTYPES(at::BFloat16)) {
 }
 #endif
 
+#define GER_CHECK_ARGVALUES(Dtype)           \
+  do {                                        \
+    CUDABLAS_NONNEGINT_CHECK(gemv<Dtype>, m); \
+    CUDABLAS_NONNEGINT_CHECK(gemv<Dtype>, n); \
+    CUDABLAS_POSINT_CHECK(gemv<Dtype>, lda);  \
+    CUDABLAS_POSINT_CHECK(gemv<Dtype>, incx); \
+    CUDABLAS_POSINT_CHECK(gemv<Dtype>, incy); \
+  } while (0)
+
+template <>
+void ger<float>(CUDABLAS_GER_ARGTYPES(float)) {
+  NoTF32Guard disable_tf32;
+  // See Note [Writing Nondeterministic Operations]
+  globalContext().alertCuBLASConfigNotDeterministic();
+  cublasHandle_t handle = at::cuda::getCurrentCUDABlasHandle();
+  _cublasAdjustLdLevel2(m, n, &lda);
+  GER_CHECK_ARGVALUES(float);
+  TORCH_CUDABLAS_CHECK(
+      cublasSger(handle, m, n, &alpha, x, incx, y, incy, a, lda));
+}
+
+template <>
+void ger<double>(CUDABLAS_GER_ARGTYPES(double)) {
+  // See Note [Writing Nondeterministic Operations]
+  globalContext().alertCuBLASConfigNotDeterministic();
+  cublasHandle_t handle = at::cuda::getCurrentCUDABlasHandle();
+  _cublasAdjustLdLevel2(m, n, &lda);
+  GER_CHECK_ARGVALUES(double);
+  TORCH_CUDABLAS_CHECK(
+      cublasDger(handle, m, n, &alpha, x, incx, y, incy, a, lda));
+}
+
+#if !defined(__HIP_PLATFORM_HCC__) || (defined(__HIP_PLATFORM_HCC__) && HIP_VERSION >= 210)
+template <>
+void ger<c10::complex<float>>(CUDABLAS_GER_ARGTYPES(c10::complex<float>)) {
+  NoTF32Guard disable_tf32;
+  // See Note [Writing Nondeterministic Operations]
+  globalContext().alertCuBLASConfigNotDeterministic();
+  cublasHandle_t handle = at::cuda::getCurrentCUDABlasHandle();
+  _cublasAdjustLdLevel2(m, n, &lda);
+  GER_CHECK_ARGVALUES(c10::complex<float>);
+  TORCH_CUDABLAS_CHECK(
+      cublasCgeru(handle, m, n, reinterpret_cast<const cuComplex*>(&alpha),
```

Comment:
calling `cublasCgeru` in place of `cublasCgerc` which transposes y instead of conjugating it. this is in line with the numpy convention https://numpy.org/doc/stable/reference/generated/numpy.outer.html